be/src/exec/broker_reader.cpp

```diff
@@ -156,24 +166,41 @@ Status BrokerReader::read(uint8_t* buf, size_t* buf_len, bool* eof) {
 
     if (response.opStatus.statusCode == TBrokerOperationStatusCode::END_OF_FILE) {
         // read the end of broker's file
-        *eof = _eof = true;
+        _eof = true;
+        *bytes_read = 0;
         return Status::OK;
     } else if (response.opStatus.statusCode != TBrokerOperationStatusCode::OK) {
         std::stringstream ss;
-        ss << "Read from broker failed, broker:" << broker_addr 
+        ss << "Read from broker failed, broker:" << broker_addr
             << " failed:" << response.opStatus.message;
         LOG(WARNING) << ss.str();
         return Status(ss.str());
     }
 
-    *buf_len = response.data.size();
-    memcpy(buf, response.data.data(), *buf_len);
-    _cur_offset += *buf_len; 
-    *eof = false;
+    *bytes_read = response.data.size();
+    memcpy(out, response.data.data(), *bytes_read);
+    _cur_offset += *bytes_read;
```

Comment:
you should set `_cur_offset` to position at the begin of this function, or there will be something wrong